src/cli/self_update/shell.rs

```diff
@@ -255,6 +256,60 @@ impl UnixShell for Fish {
     }
 }
 
+struct Nu;
+
+impl UnixShell for Nu {
+    fn does_exist(&self, process: &Process) -> bool {
+        // nu has to either be the shell or be callable for nu setup.
+        matches!(process.var("SHELL"), Ok(sh) if sh.contains("nu"))
+            || utils::find_cmd(&["nu"], process).is_some()
+    }
+
+    fn rcfiles(&self, process: &Process) -> Vec<PathBuf> {
+        let p0: Vec<_> = process
+            .var("XDG_CONFIG_HOME")
+            .ok()
+            .iter()
+            .flat_map(|p| {
+                let path = PathBuf::from(p).join("nushell/");
+                [path.join("env.nu"), path.join("config.nu")]
+            })
+            .collect();
+
+        let p1: Vec<_> = process
+            .home_dir()
+            .iter()
+            .flat_map(|p| {
+                let path = PathBuf::from(p).join(".config/nushell/");
+                [path.join("env.nu"), path.join("config.nu")]
+            })
+            .collect();
+
+        [p0, p1].concat()
+    }
+
+    fn update_rcs(&self, process: &Process) -> Vec<PathBuf> {
+        let rcs = self.rcfiles(process);
+        if rcs.len() == 4 {
+            // The first two rcfile takes precedence(XDG_CONFIG_HOME).
+            rcs[0..2].to_vec()
```

Comment:
You can use [`vec.resize(2)`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize) to save an allocation :)

Reply:
I'll fix them soon.